api/exchange.go

```diff
@@ -14,6 +14,18 @@ type ExchangeAPIKey struct {
 	Secret string
 }
 
+// CcxtParam specifies an additional ccxt parameter
+type CcxtParam struct {
```

Comment:
@Reidmcc mostly looks good. can you rename `CcxtParam` to `ExchangeParam` everywhere?
The exchange interface abstracts away the idea of ccxt / sdex / kraken from the calling code.